src/context.ts

```diff
@@ -0,0 +1,21 @@
+import { EmitterWebhookEvent as WebhookEvent, EmitterWebhookEventName as WebhookEventName } from "@octokit/webhooks";
+
+export class Context<T extends WebhookEventName = WebhookEventName> {
```

Comment:
This looks like an elegant implementation. Was this taken from Probot source code?

Reply:
Yes, I looked at how Probot implemented it and modified for our needs. Basically octokit/webhooks returns event in the callback function but they support transforming it so we can wrap the event in our own context and add additional things like octokit and bot config which I will add in later PRs.

> Seems like good code but I don't fully understand what you did related to the complex types issue.
> 
> Is this some type of type guarding via runtime event handlers?

We are basically using octokit's `on` function to bind our handlers and when you bind to specific event like `issue_comment.created` octokit's types ensure that you will get correct type in the callback.

Before we were binding to all events which caused the callback function to have union of all types and that handlers object was just too complex, but by binding to specific events we don't need `handlers` object.